File: binary_search_and_seive/CountFactors/code.js

// you can write to stdout for debugging purposes, e.g.
// console.log('this is a debug message');

function solution(N) {
    // write your code in JavaScript (Node.js 8.9.4)

    if (N === 1)
        return 1;
    let cnt = 2, i = 2;
    for (i = 2; i * i < N; i++) {
        if (N % i === 0) {
            cnt += 2;
        }
    }
    if (i * i == N)
        cnt++;
    return cnt;
}

File: Greedy/TieRopes/code.java
// you can also use imports, for example:
// import java.util.*;

// you can write to stdout for debugging purposes, e.g.
// System.out.println("this is a debug message");

class Solution {
    public int solution(int K, int[] A) {
        // write your code in Java SE 8
        int cnt = 0, sum = 0;
        for (int x : A) {
            if (x >= K) {
                sum = 0;
                cnt++;
            } else {
                sum += x;
                if (sum >= K) {
                    sum = 0;
                    cnt++;
                }
            }
        }
        return cnt;
    }
}